src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/yflows/YFlowPathSwapSpec.groovy

```diff
@@ -325,140 +267,105 @@ class YFlowPathSwapSpec extends HealthCheckSpecification {
 
         then: "Paths of the y-flow is not changed"
         Wrappers.wait(WAIT_OFFSET) {
-            verifyAll(northbound.getFlow(sFlow1.flowId)) {
+            verifyAll(northbound.getFlow(initialPath.subFlowPaths.first().flowId)) {
                 status == FlowState.UP.toString()
                 flowStatusDetails.mainFlowPathStatus == "Up"
                 flowStatusDetails.protectedFlowPathStatus == "Up"
             }
-            verifyAll(northbound.getFlow(sFlow2.flowId)) {
+            verifyAll(northbound.getFlow(initialPath.subFlowPaths.last().flowId)) {
                 status == FlowState.UP.toString()
                 flowStatusDetails.mainFlowPathStatus == "Up"
                 flowStatusDetails.protectedFlowPathStatus == "Up"
             }
         }
 
-        def sFlow1LastPathInfo = northbound.getFlowPath(sFlow1.flowId)
-        def sFlow1LastPrimary = PathHelper.convert(sFlow1LastPathInfo)
-        def sFlow1LastProtected = PathHelper.convert(sFlow1LastPathInfo.protectedPath)
-        assert sFlow1LastPrimary != sFlow1LastProtected
-        assert sFlow1InitialPrimary == sFlow1LastPrimary
-        assert sFlow1InitialProtected == sFlow1LastProtected
-
-        def sFlow2LastPathInfo = northbound.getFlowPath(sFlow2.flowId)
-        def sFlow2LastPrimary = PathHelper.convert(sFlow2LastPathInfo)
-        def sFlow2LastProtected = PathHelper.convert(sFlow2LastPathInfo.protectedPath)
-        assert sFlow2LastPrimary != sFlow2LastProtected
-        assert sFlow2InitialPrimary == sFlow2LastPrimary
-        assert sFlow2InitialProtected == sFlow2LastProtected
+        def updatedPathAfterPortUp = yFlow.retrieveAllEntityPaths()
+        updatedPathAfterPortUp.subFlowPaths.each { subFlowPath ->
+            assert subFlowPath.path.forward == updatedPathAfterPortDown.subFlowPaths.find { it.flowId == subFlowPath.flowId }.path.forward
+            assert subFlowPath.protectedPath.forward == updatedPathAfterPortDown.subFlowPaths.find { it.flowId == subFlowPath.flowId }.protectedPath.forward
+            assert subFlowPath.getCommonIslsWithProtected().isEmpty()
+        }
 
         cleanup: "Revert system to original state"
-        yFlow && yFlowHelper.deleteYFlow(yFlow.YFlowId)
+        yFlow && yFlow.delete()
         islHelper.restoreIsl(islToBreak)
-        topology.getIslsForActiveSwitches().collectMany { [it, it.reversed] }.each { database.resetIslBandwidth(it) }
+        database.resetIslsBandwidth(topology.getIslsForActiveSwitches().collectMany { [it, it.reversed] })
         database.resetCosts(topology.isls)
     }
 
     SwitchTriplet findSwitchTripletForYFlowWithProtectedPaths() {
-        return topologyHelper.switchTriplets.find {
-            def ep1paths = it.pathsEp1.unique(false) { a, b -> a.intersect(b) == [] ? 1 : 0 }
-            def ep2paths = it.pathsEp2.unique(false) { a, b -> a.intersect(b) == [] ? 1 : 0 }
+        return topologyHelper.getAllNotNeighbouringSwitchTriplets().find {
```

Comment:
getAllNotNeighbouringSwitchTriplets() includes checks it.ep1 != it.ep2 && it.ep1 != it.shared && it.ep2 != it.shared